### ISSUE TITLE
Ability to assign Discord roles based off in-game statistics 

### ISSUE THREAD

**Issue by wyverntail** [DiscordSRV/DiscordSRV#715]
Would it be possible to add a feature where a Discord role can be assigned to a player based on their in-game statistics? 

I really wanted to be able to assign a role to people who are active, to highlight them in the crowd of inactive players. Could there be a way for a role to be assigned to a member if he/she logs in, and then removed if they dont log in after a certain amount of time? or maybe just by counting the actual time spent playing the game? This could be a great way to add roles as rewards. 

I also really wanted to be able to assign a role to people based on the advancements they have completed. Whether that be a role for one advancement (such as Withering Heights, for example) or for completing all the advancements/ basic advancements/ challenge advancements. 

I feel like it'd be a really fun feature and can even be used to gain permissions to something by completing challenges that the admin can specify.

**Comment by QuinnBast** [comment]
I think this is outside of the scope of Discord SRV. If you want something like this done you'd need to make your own custom discord bot that goes through the message history and finds what you would consider to be active players.

**Comment by granny** [comment]
> I think this is outside of the scope of Discord SRV. If you want something like this done you'd need to make your own custom discord bot that goes through the message history and finds what you would consider to be active players.

They are talking about minecraft statistics, not discord statistics.

**Comment by ryantheleach** [comment]
This would be fun, But I think it could be better suited as a extra plugin.

Does DiscordSRV expose enough API to make this possible for a different plugin to implement this?

**Comment by QuinnBast** [comment]
> They are talking about minecraft statistics, not discord statistics.

Yes I know. Their ask is to assign a role to people in discord who have recently been online in the minecraft server. But DiscordSRV has no idea what statistics it should be looking at (Blocks mined? Chat messages sent? Last online?), nor does it know when to assign a role. There's just way too many variables here.

I'm not the creators of DiscordSRV, but if you want this, it would have to be a custom minecraft plugin that uses the DiscordSRV api.
